integration_test.go

```diff
@@ -0,0 +1,207 @@
+		t.Fatalf("Failed to query data information BigQuery: %v", err)
+	}
+
+	rowCount := 0
+	for {
+		//Check that the data sent is correct
+		var BQvalues []bigquery.Value
+		err := BQdata.Next(&BQvalues)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Fatalf("Failed to read query results: %v", err)
+		}
+
+		assert.Equal(t, "hello world", BQvalues[0])
+		rowCount++
+	}
+
+	// Verify the number of rows
+	assert.Equal(t, 10, rowCount)
+
+	// Clean up - delete the BigQuery table
+	if err := table.Delete(ctx); err != nil {
+		t.Fatalf("Failed to delete BigQuery table: %v", err)
+	}
+
+	// Clean up - delete the log file
+	if err := os.Remove(logFilePath); err != nil {
+		t.Fatalf("Failed to delete log file: %v", err)
+	}
+
+	// Clean up - delete the config file
+	if err := os.Remove(configFilePath); err != nil {
+		t.Fatalf("Failed to delete log file: %v", err)
+	}
+}
+
+// generate a random string with length 10 (to act as a hash for the table name)
+func randString() string {
+	source := rand.NewSource(time.Now().UnixNano())
+	rng := rand.New(source)
+	charset := "qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789"
+	str := make([]byte, 10)
+	for i := range str {
+		str[i] = charset[rng.Intn(len(charset))]
+	}
+	return string(str)
+}
+
+// template for the config file - dynamically update TableId field
+const configTemplate = `
+[SERVICE]
+    Daemon          off
+    Log_Level       error
+    Parsers_File    ./jsonparser.conf
+    plugins_file    ./plugins.conf
+[INPUT]
+    Name                               tail
+    Path                               ./logfile.log
+    Parser                             json
+    Tag                                hello_world
+[OUTPUT]
+    Name                               writeapi
+    Match                              hello_world
+    ProjectId                          bigquerytestdefault
+    DatasetId                          siddag_summer2024
+    TableId                            {{.CurrTableName}}
+`
+
+// struct for dynamically updating config file
+type Config struct {
+	CurrTableName string
+}
+
+// function creates configuration file with the input as the TableId field
+func createConfigFile(currTableID string) error {
+	//struct with the TableId
+	config := Config{CurrTableName: currTableID}
+
+	//create a new template with the format of configTemplate
+	tmpl, err := template.New("currConfig").Parse(configTemplate)
+	if err != nil {
+		return err
+	}
+
+	//create the new file
+	file, err := os.Create("./fluent-bit.conf")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	//return file with the given template and TableId
+	return tmpl.Execute(file, config)
+}
+
+//Log entry template (corresponding to BQ table schema)
+
+type log_entry struct {
+	Message string `json:"Message"`
+}
+
+// data generation function
+func generateData(numRows int) error {
+	//open file
+	file, err := os.OpenFile("logfile.log", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
```

Comment:
Can we make "logfile.log" be a parameter? That way it is not hard-coded in several places.

Reply:
Done.